src/index.ts

```diff
@@ -1,12 +1,22 @@
 import { onSnapshot, applySnapshot } from 'mobx-state-tree'
 
-import * as LocalStorage from './localStorageAdaptor.js'
+import AsyncLocalStorage from './asyncLocalStorage'
 
-export const persist = (name, store, options = {}) => {
+export interface IArgs {
+  (name: string, store: object, options?: IOptions): void
```

Comment:
Is `object` fine? I think it could be `IStateTreeNode` to be more specific but also not sure if I'm missing something and `any` is more appropriate or if `IStateTreeNode` doesn't extend `object` and maybe this is wrong

Reply:
also this should return a `Promise<void>`, woops